sdk/tables/azure-data-tables/src/samples/java/TableServiceAsyncClientCodeSnippets.java

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.util.logging.ClientLogger;
+import java.util.HashMap;
+import java.util.Map;
+import reactor.core.publisher.Mono;
+
+/**
+ * async code snippets for the table service
+ */
+public class TableServiceAsyncClientCodeSnippets {
+    final ClientLogger logger = new ClientLogger("TableServiceAsyncClientCodeSnippets");
+
+    /**
+     * create table code snippet
+     */
+    public void createTable() {
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        tableServiceAsyncClient.createTable("OfficeSupplies").subscribe(
+            Void -> {},
+            error -> logger.error("There was an error creating the table. Error: " + error),
+            () -> logger.info("Table creation successful."));
+    }
+
+    /**
+     * delete table code snippet
+     */
+    public void deleteTable() {
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        tableServiceAsyncClient.deleteTable("OfficeSupplies").subscribe(
+            Void -> {},
+            error -> logger.error("There was an error deleting the table. Error: " + error),
+            () -> logger.info("Table deletion successful."));
+    }
+
+    /**
+     * query tables code snippet
+     */
+    public void queryTable() {
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+        QueryOptions queryOptions = new QueryOptions();
```

Comment:
Highlight usage of the fluent model.

var queryOptions = new QueryOptions().setFilter("TableName eq OfficeSupplies");